### PR TITLE
Adding Dogecoin support

### PR DESCRIPTION
**Pull request by amanusk**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@ latest_rust = []  # use latest Rust features (otherwise, support Rust 1.34)
 [dependencies]
 base64 = "0.10"
 bincode = "1.0"
-bitcoin = { version = "0.23", features = ["use-serde"] }
+bitcoin = { path = "../rust-bitcoin", features = ["use-serde"] }
```
Comment (oleiba) [inline Cargo.toml]: I think we'll want rust-bitcoin to be a fork in the ZenGo-X account, and to be used within a specific branch.

File: src/config.rs [hunk 2]
```diff
@@ -198,16 +200,22 @@ impl Config {
...
             Network::Testnet => 18332,
             Network::Regtest => 18443,
+            Network::Dogecoin => 22555,
+            Network::Dogetest => 44555,
         };
         let default_electrum_port = match config.network {
             Network::Bitcoin => 50001,
             Network::Testnet => 60001,
             Network::Regtest => 60401,
+            Network::Dogecoin => 60001,
```
Comment (oleiba) [inline src/config.rs]: Let's use 50001 to be consistent with electrs testnet

File: src/daemon.rs [hunk 3]
```diff
@@ -22,8 +22,12 @@ use crate::errors::*;
 use crate::metrics::{HistogramOpts, HistogramVec, Metrics};
 use crate::signal::Waiter;
 use crate::util::HeaderList;
+use bitcoin::consensus::encode::Decodable;
 use bitcoin::BitcoinHash;
 
+// #[macro_use]
+// use crate::js_try;
+
```
Comment (oleiba) [inline src/daemon.rs:31]: Delete?

File: src/daemon.rs [hunk 4]
```diff
@@ -33,21 +37,57 @@ fn parse_hash<T: Hash>(value: &Value) -> Result<T> {
...
     let block_bytes = hex::decode(block_hex).chain_err(|| "non-hex block")?;
-    Ok(deserialize(&block_bytes).chain_err(|| format!("failed to parse block {}", block_hex))?)
+
+    let mut cur = std::io::Cursor::new(&block_bytes);
+    let block_header: bitcoin::BlockHeader =
+        Decodable::consensus_decode(&mut cur).chain_err(|| "Unable to decode header")?;
+
+    match network {
+        Network::Dogecoin | Network::Dogetest if block_header.version & 1 << 8 != 0 => {
+            let _: bitcoin::Transaction =
```
Comment (oleiba) [inline src/daemon.rs:78]: 1. Please extract this into a helper function
2. Use constants for magic numbers
3. Add documentation for when was this logic taken from, and where this ideally should be put in (rust-bitcoin)?

File: src/daemon.rs [hunk 5]
```diff
@@ -473,8 +513,14 @@ impl Daemon {
     }
 
     pub fn estimatesmartfee(&self, conf_target: usize, estimate_mode: &str) -> Result<f64> {
-        let val: Value = self.request("estimatesmartfee", json!([conf_target, estimate_mode]))?;
-        let fee_estimate: Result<FeeEstimate> = Ok(from_value(val).chain_err(|| "invalid fee estimate")?);
+        let val: Value;
+        if self.network == Network::Dogecoin || self.network == Network::Dogetest {
+            val = self.request("estimatesmartfee", json!([conf_target]))?;
```
Comment (oleiba) [inline src/daemon.rs:533]: See if we can make a reliable fee estimation, as you said it's x5 the cost.

File: src/daemon.rs [hunk 6]
```diff
@@ -498,28 +547,41 @@ impl Daemon {
...
-        self.request("getblock", json!([blockhash.to_hex(), /*verbose=*/ 1]))?
-            .get("tx")
-            .chain_err(|| "block missing txids")?
-            .as_array()
-            .chain_err(|| "invalid block txids")?
-            .iter()
-            .map(parse_hash)
-            .collect::<Result<Vec<Txid>>>()
+        if self.network == Network::Dogecoin || self.network == Network::Dogetest {
+            self.request("getblock", json!([blockhash.to_hex(), /*verbose=*/ true]))?
```
Comment (oleiba) [inline src/daemon.rs:582]: Maybe for readability - 
make the `if self.network == Network::Dogecoin || self.network == Network::Dogetest` condition only for the verbose value?...

File: src/daemon.rs [hunk 7]
```diff
@@ -535,7 +597,9 @@ impl Daemon {
         let values = self.requests("getblock", &params_list)?;
         let mut blocks = vec![];
         for value in values {
-            blocks.push(block_from_value(value)?);
+            let block = block_from_value(value, self.network)?;
+            trace!("get block {}", block.bitcoin_hash());
```
Comment (oleiba) [inline src/daemon.rs]: In the end - let's get rid of all the `trace!` prints

File: src/lib.rs [hunk 8]
```diff
@@ -19,11 +19,12 @@ pub mod daemon;
 pub mod errors;
 pub mod fake;
 pub mod index;
+// pub mod macs;
```
Comment (oleiba) [inline src/lib.rs]: delete?

File: src/daemon.rs (hunk 6 above)
Comment (amanusk) [reply src/daemon.rs:582]: Not sure I can define a variable that will have a different type based on the case in rust. Can I?

**Pull request closed without merging**